src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```
@@ -32,25 +35,67 @@ public PublicationService(ArtefactRepository artefactRepository,
         artefact.setSearch(payloadExtractor.extractSearchTerms(payload));
 
         return artefactRepository.save(artefact);
     }
+
+
+    /**
+     * Get all relevant artefacts relating to a given court ID.
+     *
+     * @param searchValue - represents the court ID in question being searched for
+     * @param verified    - represents the verification status of the user. Currently only verified/non-verified, but
+     *                    will include other verified user types in the future
+     * @return a list of all artefacts that fulfil the timing criteria, match the given court id and sensitivity
+     *                    associated with given verification status
+     */
+    public List<Artefact> findAllByCourtId(String searchValue, Boolean verified) {
+        LocalDateTime currDate = LocalDateTime.now();
+        if (verified) {
+            return artefactRepository.findArtefactsBySearchVerified(searchValue, currDate);
+        } else {
+            return artefactRepository.findArtefactsBySearchUnverified(searchValue, currDate);
+        }
+
+    }
+
+    /**
+     * takes in provenance and artefact id and returns the data within the matching blob in string format.
+     * @param artefactId represents the artefact id which is then used to get an artefact to populate the inputs
+     *                   for the blob request
+     * @return the data within the blob in string format
+     */
+    public String getByArtefactId(UUID artefactId) {
+        Optional<Artefact> optionalArtefact = artefactRepository.findByArtefactId(artefactId);
+        if (optionalArtefact.isPresent()) {
+            Artefact artefact;
+            artefact = optionalArtefact.get();
+            String sourceArtefactId = artefact.getSourceArtefactId();
+            String provenance = artefact.getProvenance();
+            return azureBlobService.getBlobData(sourceArtefactId, provenance);
+        } else {
+            return azureBlobService.getBlobData("", "");
```

Is this needed / can we just throw an exception here instead?

---

To be honest, I just did this so as not to break the existing tests. Absolutely can just throw an exception, yeah :)